packages/truffle/test/scenarios/external_compilers/truffle-compile.js

```diff
@@ -0,0 +1,99 @@
+var MemoryLogger = require("../memorylogger");
+var CommandRunner = require("../commandrunner");
+var contract = require("truffle-contract");
+var fs = require("fs");
+var path = require("path");
+var assert = require("assert");
+var sandbox = require("../sandbox");
+var Server = require("../server");
+var Reporter = require("../reporter");
+
+describe("`truffle compile` as external", function() {
+  var config;
+  var project = path.join(__dirname, '../../sources/external_compile');
+  var logger = new MemoryLogger();
+
+  before("set up the server", function(done) {
+    Server.start(done);
+  });
+
+  after("stop server", function(done) {
+    Server.stop(done);
+  });
+
+  before("set up sandbox", function() {
+    this.timeout(10000);
+    return sandbox.create(project).then(conf => {
+      config = conf;
+      config.network = "development";
+      config.logger = logger;
+      config.mocha = {
+        reporter: new Reporter(logger)
+      }
+    });
+  });
+
+  it("will compile", function(done) {
+    this.timeout(20000);
+
+    CommandRunner.run("compile --compiler=external", config, function(err) {
```

Comment:
I like this test. Without this flag compile will run both solc compilation *and* whatever is specified by external?

Reply:
Correct